src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,179 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    private T unlink(Node<T> x) {
+        T element = x.element;
+        Node<T> next = x.next;
+        Node<T> prev = x.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            x.prev = null;
+        }
+
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            x.next = null;
+        }
+
+        x.element = null;
+        size--;
+        return element;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> tmp = last;
+        Node<T> newNode = new Node<>(last, value, null);
+        last = newNode;
+        if (tmp == null) {
+            first = newNode;
+        } else {
+            tmp.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            final Node<T> pred = getNode(index).prev;
+            final Node<T> newNode = new Node<>(pred, value, getNode(index));
+            getNode(index).prev = newNode;
+            if (pred == null) {
+                first = newNode;
+            } else {
+                pred.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
```

Comment:
<img width="807" alt="Screen Shot 2023-01-04 at 9 25 38 PM" src="https://user-images.githubusercontent.com/90692308/210633636-57ccfacf-104f-457f-88bf-38bad10ce9c7.png">
Not fixed